lambdas/credentialissuerreturn/src/main/java/uk/gov/di/ipv/core/credentialissuerreturn/CredentialIssuerReturnHandler.java

```diff
@@ -190,7 +191,11 @@ private Optional<ErrorResponse> validate(CredentialIssuerRequestDto request) {
             return Optional.of(ErrorResponse.MISSING_OAUTH_STATE);
         }
 
-        if (!request.getState().equals(getPersistedOauthState(request))) {
+        String persistedOauthState = getPersistedOauthState(request);
+        if (!request.getState().equals(persistedOauthState)) {
+            LoggingUtils.appendKey("expected_state", persistedOauthState);
+            LoggingUtils.appendKey("request_state", request.getState());
```

Comment:
Is this a place where a structured messsage would be better than attaching the value to the loggger?

Reply:
I think so - that's why I've appended the state values as fields with `appendKey()`